### PR TITLE
Provide a way to determine if a durable object is broken from errors

### PR DESCRIPTION
**Pull request by bcaimano**
Currently, workers have to rely upon the `Error.message` value to determine if their durable object is broken. Usually, the worker will want to retry their request to reinitialize their durable object and continue work. Thus we want to set a property like the existing `Error.remote` property to convey that the the request failed because the durable object became broken.

### CODE REVIEW

File: src/workerd/jsg/util.c++ [hunk 1]
```diff
@@ -178,45 +193,50 @@ TunneledErrorType tunneledErrorType(kj::StringPtr internalMessage) {
     }
   };
 
-  auto tryExtractError =
-      [isFromRemote](kj::StringPtr msg) -> kj::Maybe<TunneledErrorType> {
+  auto tryExtractError = [isFromRemote, &isDurableObjectBroken](kj::StringPtr msg)
```
Comment (MellowYarker) [inline src/workerd/jsg/util.c++:196]: nit: Since we aren't modifying `isDurableObjectBroken` and it's a primitive, it feels like we should pass a value.

**Review by MellowYarker — APPROVED** [review]
Thanks for doing this, I remember R2 had mentioned wanting this!

File: src/workerd/jsg/util.c++ (hunk 1 above)
Comment (bcaimano) [reply src/workerd/jsg/util.c++:196]: Ahhhh, good point! And it should all inline better if the lambda only has arguments anyway.

File: src/workerd/jsg/util.c++ [hunk 2]
```diff
@@ -123,6 +123,16 @@ bool setRemoteError(v8::Isolate* isolate, v8::Local<v8::Value>& exception) {
       v8::True(isolate)));
 }
 
+bool setDurableObjectResetError(v8::Isolate* isolate, v8::Local<v8::Value>& exception) {
+  KJ_ASSERT(exception->IsObject());
+  auto obj = exception.As<v8::Object>();
+  return jsg::check(
+    obj->Set(
+      isolate->GetCurrentContext(),
+      jsg::v8Str(isolate, "isDurableObjectReset"_kj, v8::NewStringType::kInternalized),
```
Comment (a-robinson) [inline src/workerd/jsg/util.c++:132]: The `is` prefix here seems a bit inconsistent with the `.remote` field (i.e. it's `.remote`, not `.isRemote`). I don't feel super strongly, but I think just `.durableObjectReset` would match better.

**Review by a-robinson — APPROVED** [review]

File: src/workerd/jsg/util.c++ (hunk 2 above)
Comment (bcaimano) [reply src/workerd/jsg/util.c++:132]: Hmmmmm, I'd be curious as to @jasnell 's opinion here. I considered naming parity with `.remote` but `.durableObjectReset` didn't feel like a property name for a boolean to me.

Comment (jasnell) [reply src/workerd/jsg/util.c++:132]: I agree with @a-robinson here. I'm not a fan of the `is*` prefix on getters for JavaScript types.

**Review by jasnell — APPROVED** [review]

File: src/workerd/jsg/util.c++ (hunk 2 above)
Comment (bcaimano) [reply src/workerd/jsg/util.c++:132]: I'll change it over!

**Pull request merged**